core/src/kmx/kmx_plus.cpp

```diff
@@ -375,11 +376,100 @@ COMP_KMXPLUS_TRAN::valid(KMX_DWORD _kmn_unused(length)) const {
     assert(false);
     return false;
   }
-  // TODO-LDML
-  DebugLog("!! More to do here.");
   return true;
 }
 
+COMP_KMXPLUS_TRAN_Helper::COMP_KMXPLUS_TRAN_Helper()
+    : tran(nullptr), is_valid(false), groups(nullptr), transforms(nullptr), reorders(nullptr) {
+}
+
+bool
+COMP_KMXPLUS_TRAN_Helper::setTran(const COMP_KMXPLUS_TRAN *newTran) {
+  if (newTran != nullptr) {
+    // Only print this if not null.
+    DebugLog("tran helper: validating '%c%c%c%c' newTran=%p", DEBUG_IDENT(newTran->header.ident), newTran);
+  }
+  is_valid = true;
+  if (newTran == nullptr) {
+    DebugLog("tran helper: invalid, newTran=%p", newTran);
+    // null = invalid
+    is_valid = false;
+    // No assert here: just a missing layer
+    return false;
+  }
+  tran = newTran;
+  const uint8_t *rawdata = reinterpret_cast<const uint8_t *>(tran);
+  rawdata += LDML_LENGTH_TRAN;  // skip past non-dynamic portion
```

Comment:
Are we validating that we aren't pointing or reading outside buffers anywhere?

Reply:
Yes the validity check on the buffer size happens in …TRAN.valid

Reply:
So this can't be called before TRAN.valid? Just wanting to make sure we don't accidentally pass bogus data to this function.

Reply:
"A programmer looks both ways on a one way street". 

It wouldn't hurt to call newTran.valid() although I'm remembering that requires the end of memory region info which we don't have here.

Reply:
TRAN.valid() is called by the macro that initialized the "tran" field in the kmx_plus object. And there it's called with the end of memory pointer to validate the end. So there shouldn't be a path where an invalid tran is passed to this function.

Reply:
> "A programmer looks both ways on a one way street".

In this part of the world, everyone looks both ways on a one way street. Or doesn't look at all.